docs(README): Tweak formatting, punctuation for clarity (#2465)
tj/commander.js: Readme.md

## Options

Options are defined with the `.option()` method, also serving as documentation for the options. Each option can have a short flag (single character) and a long name, separated by a comma, a space, or a vertical bar (`|`). 

> 

Reviewer: The blank line and `Examples:` may make it seem the example will be of short-ish flags. If you think the blank line is helpful, perhaps remove the `Examples:` to avoid that impression.

To allow a wider range of short-ish flags than just

> single characters, you may also have two long options. Examples:

Reviewer: As above, perhaps:
```suggestion
single characters, you may also have two long options.
```

The parsed options can be accessed by calling `.opts()` on a `Command` object, and are passed to the action handler.

> Multi-word options, such as `"--template-engine"`, are camel-cased, becoming `program.opts().templateEngine`, for example.

Reviewer: I think this is an old line with tweaks over the years! Still clunky. How about a bigger rewrite:

Multi-word options like `--template-engine` are normalized to camelCase option names, resulting in properties such as `program.opts().templateEngine`.

Author: Suggest the change and I’ll give it a ✅

Author: never mind I’ve got a few minutes to do the change :)

## Commands

You can specify (sub)commands using `.command()` or `.addCommand()`. There are two ways these can be implemented:

1. Using an action handler attached to the command; or,

> 2. as a stand-alone executable file. (More detail about this later.)

Reviewer: I don't think this expansion is enough of an improvement over the concise statement.

Author: K.

> Subcommands may be nested. (See example: [nestedCommands.js](./examples/nestedCommands.js).)

Reviewer: I settled on `Example file:` elsewhere, and I think use that here too.

```suggestion
Subcommands may be nested. Example file: [nestedCommands.js](./examples/nestedCommands.js).
```

Configuration options can be passed with the call to `.command()` and `.addCommand()`. Specifying `hidden: true` will
remove the command from the generated help output. Specifying `isDefault: true` will run the subcommand if no other

> subcommand is specified. (See example: [defaultCommand.js](./examples/defaultCommand.js).)

Reviewer: ```suggestion
subcommand is specified. (Example file: [defaultCommand.js](./examples/defaultCommand.js).)
```

> You can add alternative names for a command with `.alias()`. (See example: [alias.js](./examples/alias.js).)

Reviewer: ```suggestion
You can add alternative names for a command with `.alias()`. (Example file: [alias.js](./examples/alias.js).)
```

`.command()` automatically copies the inherited settings from the parent command to the newly created subcommand. This is only done during creation; any later setting changes to the parent are not inherited.

For safety, `.addCommand()` does not automatically copy the inherited settings from the parent command. There is a helper routine `.copyInheritedSettings()` for copying the settings when they are wanted.

### Command-arguments

For subcommands, you can specify the argument syntax in the call to `.command()` (as shown above). This
is the only method usable for subcommands implemented using a stand-alone executable.

> For other subcommands, you can instead use the following method.

Reviewer: The "For other subcommands" is in contrast to "implemented using a stand-alone executable" which gets lost with the line break. 

I think this needs a bigger rewrite. Want to have a go? The background is that `.argument()` is newer and somewhat preferred, but you can't use it with stand-alone executables (because there isn't any opportunity for chaining calls).

Author: Alas, I’m out of time today. You’re welcome to resolve the rewrite as you like! 🙏🏽

Author: I squeezed in an attempted edit.

  });
```

The last argument of a command can be variadic, and _only_ the last argument.  To make an argument variadic, simply
append `...` to the argument name.

> A variadic argument is passed to the action handler as an array. For example:

Reviewer: I am neutral on the blank line. If you do like it, I suggest dropping the "for example" as example is for the whole thing.

```suggestion
A variadic argument is passed to the action handler as an array.
```

  });
```

> If you prefer, you can work with the command directly and skip declaring the parameters for the action handler. The `this` keyword is set to the running command, and can be used from a function expression (but not from an arrow function).

Reviewer: I wanted one clause and not two. ChatGPT agreed with you that comma needed as written, but suggested a rewrite to allow no comma:

```suggestion
If you prefer, you can work with the command directly and skip declaring the parameters for the action handler. The `this` keyword is set to the running command and usable from a function expression (but not from an arrow function)..
```

Author: ChatGPT? Meh.  I think I got a better  rewrite. Just committed.

### TypeScript

> **extra-typings:** There is an optional project to infer extra type information from the option and argument definitions.

Reviewer: Ok, but I suspect my markdown linter is going to complain about using bold as headings. 😆

Author: I don’t think of it like a heading.  I think of it more like a [`dl` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dl).  

Tell the linter who’s boss. 😈

const program = createCommand();
```

`createCommand()` is also a method of the `Command` object, and creates a new command rather than a subcommand. This gets used internally
when creating subcommands using `.command()`, and you may override it to

> customise the new subcommand. See example file: [custom-command-class.js](./examples/custom-command-class.js).

Reviewer: ```suggestion
customise the new subcommand. Example file: [custom-command-class.js](./examples/custom-command-class.js).
```

### Node options such as `--harmony`

You can enable `--harmony` option in two ways:

> - Use `#! /usr/bin/env node --harmony` in the subcommands scripts. (**Note:** Windows does not support this pattern.)

Reviewer: I don't like the bold here, too noisy.

```suggestion
- Use `#! /usr/bin/env node --harmony` in the subcommands scripts. (Note: Windows does not support this pattern.)
```

If you are using the node inspector for [debugging](https://nodejs.org/en/docs/guides/debugging-getting-started/) executable subcommands using `node --inspect` et al.,
the inspector port is incremented by 1 for the spawned subcommand.

If you are using VSCode to debug executable subcommands you need to set the `"autoAttachChildProcesses": true` flag in your `launch.json` configuration.

### npm run-script

By default, when you call your program using `run-script`, `npm` will parse any options on the command-line and they will not reach your program. Use
 `--` to stop the npm option parsing and pass through all the arguments.

>  The synopsis for [`npm run-script`](https://docs.npmjs.com/cli/v9/commands/npm-run-script) explicitly shows the `--` for this reason:

Reviewer: Side note: I had left out the code formatting in a couple of places you noticed as I thought it might make the link less clear, but looks fine.

### Override exit and output handling

By default, Commander calls `process.exit()` when it detects errors, or after displaying the help or version. You can override
this behaviour and optionally supply a callback. The default override throws a `CommanderError`.

The override callback is passed a `CommanderError` with the properties 
- `exitCode` (number), 
- `code` (string), and 

> - `message` (string).

Reviewer: I went compact on purpose, but agree list is clearer. I think not as a sentence though.

```suggestion
The override callback is passed a `CommanderError` with the properties:
- `exitCode`: number
- `code`: string 
- `message`: string
```
